R/app_utilities.R

```diff
@@ -1,370 +1,374 @@
+tsy_palette <- c( 
+  rgb(0, 131, 172, maxColorValue = 255),
+  rgb(0, 188, 226, maxColorValue = 255),
+  rgb(103, 168, 84, maxColorValue = 255),
+  rgb(188, 214, 81, maxColorValue = 255),
+  rgb(241, 164, 45, maxColorValue = 255),
+  rgb(239, 150, 108, maxColorValue = 255),
+  rgb(0, 79, 103, maxColorValue = 255),
+  rgb(0, 113, 136, maxColorValue = 255),
+  rgb(62, 101, 50, maxColorValue = 255),
+  rgb(122, 143, 34, maxColorValue = 255),
+  rgb(13, 143, 34, maxColorValue = 255),
+  rgb(122, 42, 34, maxColorValue = 255)
+)
+
 # Converts a string used to input children's ages into numeric values
 convert_ages <- function(input_string){
   ages = as.numeric(strsplit(input_string, ",")[[1]])
   ages = sort(ages)
   return(ages)
 }
 
-
-# Data table containing all the parameter setting for adjustment from the app
-show_all_parameters <- function(Parameters_File_SQ, Parameters_File_Reform) {
-  Params_text_SQ <- read.xlsx(Parameters_File_SQ)
-  Params_text_Reform <- read.xlsx(Parameters_File_Reform)
+calculate_income <- function(
+    params,
+    max_hours, hourly_wage, children_ages,
+    partnered, partner_wages, partner_hours,
+    accommodation_type, as_accommodation_costs, as_area,
+    steps_per_dollar = 1L, weeks_in_year = 52L

```

Comment:
A minor comment - I see all the "weeks_in_year" are hardcoded to 52; I remember we have a function that is able to adjust this when it's a leap year. (I don't think this is a major issue, but leave this comment for your consideration)

Reply:
Let's leave the weeks-in-year issue for now, the function already deals with it a certain way and we may need to think a bit about what we actually want to expose to the user of the function.